src/sequence_statistics.py

```diff
@@ -0,0 +1,229 @@
+                words += word
+                sentence += word + " "
+                pronunciation += word_pronunciation + " "
+                current = possible(space['following_ipa_syllable'])
+                i += 1
+            else:
+                print('Mot ', existence[0]['spelling'],' (', word_pronunciation, ') retrouvé')
+                current = possible(space['following_ipa_syllable'])
+    print(sentence)
+    print(sentence.replace('-',''))
+    print(pronunciation)
+    return sentence
+
+
+def generate_new_up(language,length=10):
+    space = requests.get(f'{API_URL}/{language}_syllables/espace/orthographic').json()['result']
+    current = possible(space['preceding_ipa_syllable'])
+    sentence = ""
+    pronunciation = ""
+    words = []
+    i = 0
+    while i < length:
+        word = ""
+        word_pronunciation = ""
+        while current != " ":
+            try:
+                syll = requests.get(f'{API_URL}/{language}_syllables/{current}').json()['result']
+                word = syll['orthographical_syllable'] + "-" + word
+                word_pronunciation = syll['ipa_syllable'] + word_pronunciation
+                current = possible(syll['preceding_ipa_syllable'])
+            except KeyError:
+                print('Problème pour la syllabe : ', current)
+                current = " "
+        if word not in words:
+            existence = requests.get(f'{API_URL}/{language}/{word_pronunciation}/phonetic').json()['result']
+            if existence == []:
+                words += word
+                sentence = word + " " + sentence
+                pronunciation = word_pronunciation + " " + pronunciation
+                current = possible(space['preceding_ipa_syllable'])
+                i += 1
+            else:
+                print('Mot ', existence[0]['spelling'],' (', word_pronunciation, ') retrouvé')
+                current = possible(space['preceding_ipa_syllable'])
+    print(sentence)
+    print(sentence.replace('-',''))
+    print(pronunciation)
+    return sentence
+
+
+# Retourne une syllabe possible parmi les syllabes de following, mais omet celles de used :
+# possible({'mon': 1, 'ton' : 4, 'son': 3, 'tan': 2}, ['mon','ton']) retourne 'son' avec une probabilité de 0.6 et
+# 'tan' avec une probabilité de 0.4
+def possible(following, used=[]):
+    possibilities = []
+    for key in following.keys():
+        if key not in used:
+            for i in range(following[key]):
+                possibilities += [key]
+    number = random.randint(0,len(possibilities)-1)
+    return possibilities[number]
+
+
+# Renvoie l'élément de  dic avec le score le plus grand dont la clé ne se trouve pas dans string
+# probable({'a': 1, 'b': 2, 'c': 3}, 'cet'} retourne b
+def probable(dic, string=""):
+    max = 0
+    result = ""
+    for syllable in dic.keys():
+        if (syllable not in string) and (dic[syllable] > max):
+            result = syllable
+    return result
+
+
+if __name__ == '__main__':
```

Comment:
this is good :) but maybe you could put this small code in another script ? it's code to separate the code itself from the script that runs it.